diff --git a/dwm/dwm.c b/dwm/dwm.c
index ba9bafe..be110c7 100644
--- a/dwm/dwm.c
+++ b/dwm/dwm.c
@@ -1539,8 +1539,8 @@ monocle(Monitor *m)
 			n++;
 	if (n > 0) /* override layout symbol */
 		snprintf(m->ltsymbol, sizeof m->ltsymbol, "[%d]", n);
-	for (c = nexttiled(m->clients); c; c = nexttiled(c->next))
-		resize(c, m->wx, m->wy, m->ww - 2 * c->bw, m->wh - 2 * c->bw, 0);
+    for (c = nexttiled(m->clients); c; c = nexttiled(c->next))
+        resize(c, m->wx + m->gappov, m->wy + m->gappoh, m->ww - 2 * c->bw - 2 * m->gappov, m->wh - 2 * c->bw - 2 * m->gappoh, 0);
 }
 
 void
